Source/WebCore/Modules/url-pattern/URLPattern.cpp

```diff
@@ -258,8 +258,13 @@ ExceptionOr<Ref<URLPattern>> URLPattern::create(ScriptExecutionContext& context,
 ExceptionOr<Ref<URLPattern>> URLPattern::create(ScriptExecutionContext& context, std::optional<URLPatternInput>&& input, URLPatternOptions&& options)
```

Comment:
This whole PR seems fishy. The spec says the input should be `URLPatternInput`, not `std::optional<URLPatternInput>`, there is no std::nullopt state to deal with. Our implementation should either:
- Take in a `URLPatternInput&&`
or if the generated bindings don't support that somehow, 
- keep taking a `std::optional<URLPatternInput>&&` but treat the `std::nullopt` as if the JS passed a null String.

Reply:
The spec says that there should be two `create` functions.

Reply:
>  ut treat the std::nullopt as if the JS passed a null String.

On Chrome, the behavior is to wildcard every URLPattern component. 

<img width="1571" alt="Screenshot 2024-12-23 at 5 51 41 PM" src="https://github.com/user-attachments/assets/bb499a40-b4aa-416c-a93a-904b8f03d3bd" />

Reply:
It is kind of confusing though. 

The spec for create (https://urlpattern.spec.whatwg.org/#url-pattern-create) mentions taking in an input but NOT an **optional** input. But it does mention that everything needs to be wild-carded. 
<img width="897" alt="Screenshot 2024-12-23 at 6 04 35 PM" src="https://github.com/user-attachments/assets/b8f68e2c-be21-43e3-9229-6ff6bf7ddb76" />

The spec also does not mention passing in "nothing" even though the test case clearly does take in "nothing" inputs. 
<img width="1177" alt="Screenshot 2024-12-23 at 6 03 33 PM" src="https://github.com/user-attachments/assets/77ec5dfe-9f21-4fb7-af3d-48afc2a88770" />

Reply:
The spec doesn't deal with no inputs because no inputs in not a thing, what you get is a default constructed `URLPatternInput` which I expect would be a null `USVString` (since `USVString` is the type that comes first in the IDL union).

So I think that in the spec, that means we'd exercise step 2:
> If input is a [scalar value string](https://infra.spec.whatwg.org/#scalar-value-string) then:

Then Step 5 would populate the empty `init` with wild cards.

Reply:
Note that the reason you get a default constructed `URLPatternInput` and not a `std::nullopt` is because of `= { }` in the IDL.